src/perception/RcnnMopedModule.cpp

```diff
@@ -0,0 +1,199 @@
+#include <aikido/perception/RcnnMopedModule.hpp>
+
+#include <Eigen/Geometry>
+#include <dart/utils/urdf/DartLoader.hpp>
+#include <ros/ros.h>
+#include <ros/topic.h>
+#include <visualization_msgs/Marker.h>
+#include <visualization_msgs/MarkerArray.h>
+#include <aikido/io/CatkinResourceRetriever.hpp>
+#include <aikido/perception/shape_conversions.hpp>
+
+using dart::dynamics::SkeletonPtr;
+
+namespace aikido {
+namespace perception {
+
+RcnnPoseModule::RcnnPoseModule(
+    ros::NodeHandle nodeHandle,
+    std::string markerTopic,
+    std::shared_ptr<ObjectDatabase> configData,
+    std::shared_ptr<aikido::io::CatkinResourceRetriever> resourceRetriever,
+    std::string referenceFrameId,
+    dart::dynamics::Frame* referenceLink)
+  : mNodeHandle(std::move(nodeHandle))
+  , mMarkerTopic(std::move(markerTopic))
+  , mConfigData(std::move(configData))
+  , mResourceRetriever(std::move(resourceRetriever))
+  , mReferenceFrameId(std::move(referenceFrameId))
+  , mReferenceLink(std::move(referenceLink))
+  , mTfListener(mNodeHandle)
+{
+  // Do nothing
+}
+
+bool RcnnPoseModule::detectObjects(
+    const aikido::planner::WorldPtr& env,
+    ros::Duration timeout,
+    ros::Time timestamp)
+{
+  // Checks detected objects, looks up the database,
+  // and adds new skeletons to the world env
+  bool any_valid = false;
+  visualization_msgs::MarkerArrayConstPtr marker_message
+      = ros::topic::waitForMessage<visualization_msgs::MarkerArray>(
+          mMarkerTopic, mNodeHandle, timeout);
+
+  // Making sure the Message from RcnnPose is non empty
+  if (marker_message == nullptr)
+  {
+    dtwarn << "[RcnnPoseModule::detectObjects] nullptr Marker Message "
+           << mMarkerTopic << std::endl;
+    return false;
+  }
+
+  if (marker_message->markers.size() == 0)
+  {
+    dtwarn << "[RcnnPoseModule::detectObjects] No markers on topic "
+           << mMarkerTopic << std::endl;
+    return false;
+  }
+
+  dart::utils::DartLoader urdf_loader;
+
+  for (const auto& marker_transform : marker_message->markers)
+  {
+    const auto& marker_stamp = marker_transform.header.stamp;
+    const auto& obj_key = marker_transform.text;
+    const auto& obj_id = marker_transform.ns;
+    const auto& detection_frame = marker_transform.header.frame_id;
+    if (!timestamp.isValid() || marker_stamp < timestamp)
+    {
+      continue;
+    }
+
+    // If obj_key is "None",
```

Comment:
When would `obj_key` be None? Does the RCNN keep track of that somehow?

Reply:
`RcnnPose` will set `obj_key` None when an object disappears on its list.